registration-authority.go

```diff
@@ -81,26 +82,32 @@ func fingerprint256(data []byte) string {
 	return b64enc(d.Sum(nil))
 }
 
-func (ra *RegistrationAuthorityImpl) NewAuthorization(request Authorization, key jose.JsonWebKey) (Authorization, error) {
-	zero := Authorization{}
+func lastPathSegment(url AcmeURL) string {
+	return regexp.MustCompile("^.*/").ReplaceAllString(url.Path, "")
```

Comment:
Should really save the result of MustCompile someplace for reuse, instead of recompiling it each time.